consensus/istanbul/core/core.go

```diff
@@ -462,81 +462,102 @@ func (c *core) getPreprepareWithRoundChangeCertificate(round *big.Int) (*istanbu
 	return request, roundChangeCertificate, nil
 }
 
-// startNewRound starts a new round. if round equals to 0, it means to starts a new sequence
+// startNewRound starts a new round with the desired round
 func (c *core) startNewRound(round *big.Int) error {
-	roundChange := false
+	logger := c.newLogger("func", "startNewRound", "tag", "stateTransition")
+	_, headAuthor := c.backend.GetCurrentHeadBlockAndAuthor()
+
+	if round.Cmp(c.current.Round()) == 0 {
+		logger.Trace("Already in the desired round.")
+		return nil
+	} else if round.Cmp(c.current.Round()) < 0 {
+		logger.Warn("New round should not be smaller than current round", "new_round", round)
+		return nil
+	}
+
+	// Generate next view and preprepare
+	newView := &istanbul.View{
+		Sequence: new(big.Int).Set(c.current.Sequence()),
+		Round:    new(big.Int).Set(round),
+	}
+
+	var err error
+	request, roundChangeCertificate, err := c.getPreprepareWithRoundChangeCertificate(round)
+	if err != nil {
+		logger.Error("Unable to produce round change certificate", "err", err, "new_round", round)
+		return nil
+	}
+
+	// Calculate new propose
+	valSet := c.current.ValidatorSet()
+	nextProposer := c.selectProposer(valSet, headAuthor, newView.Round.Uint64())
```

Comment:
`headAuthor` here doesn't seem right.  If we're on sequence number 100, and in the meantime we've received block 100, shouldn't we still use the author from block 99?

Reply:
If so, I see there's a function called `AuthorForBlock()` that could be used, and it could be right here rather than at the beginning of the function.

Reply:
Yea, switch it to pulling the author based on the current sequence number.